packages/datadog-plugin-aws-sdk/src/services/bedrockruntime.js

```diff
@@ -0,0 +1,207 @@
+    const textAndResponseReasons = extractTextAndResponseReason(response, modelProvider, modelName, shouldSetChoiceIds)
+
+    tags = buildTagsFromParams(requestParams, textAndResponseReasons, modelProvider, modelName, operation)
+
+    tags = Object.fromEntries(Object.entries(tags).filter(([_, v]) => v !== undefined))
+
+    return tags
+  }
+}
+
+function extractRequestParams (params, provider) {
+  const requestBody = JSON.parse(params.body)
+  const modelId = params.modelId
+
+  if (provider === _AI21) {
+    return {
+      prompt: requestBody.prompt,
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.topP || '',
+      max_tokens: requestBody.maxTokens || '',
+      stop_sequences: requestBody.stopSequences || []
+    }
+  } else if (provider === _AMAZON && modelId.includes('embed')) {
+    return { prompt: requestBody.inputText }
+  } else if (provider === _AMAZON) {
+    const textGenerationConfig = requestBody.textGenerationConfig || {}
+    return {
+      prompt: requestBody.inputText,
+      temperature: textGenerationConfig.temperature || '',
+      top_p: textGenerationConfig.topP || '',
+      max_tokens: textGenerationConfig.maxTokenCount || '',
+      stop_sequences: textGenerationConfig.stopSequences || []
+    }
+  } else if (provider === _ANTHROPIC) {
+    const prompt = requestBody.prompt || ''
+    const messages = requestBody.messages || ''
+    return {
+      prompt: prompt || messages,
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.top_p || '',
+      top_k: requestBody.top_k || '',
+      max_tokens: requestBody.max_tokens_to_sample || '',
+      stop_sequences: requestBody.stop_sequences || []
+    }
+  } else if (provider === _COHERE && modelId.includes('embed')) {
+    return {
+      prompt: requestBody.texts,
+      input_type: requestBody.input_type || '',
+      truncate: requestBody.truncate || ''
+    }
+  } else if (provider === _COHERE) {
+    return {
+      prompt: requestBody.prompt,
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.p || '',
+      top_k: requestBody.k || '',
+      max_tokens: requestBody.max_tokens || '',
+      stop_sequences: requestBody.stop_sequences || [],
+      stream: requestBody.stream || '',
+      n: requestBody.num_generations || ''
+    }
+  } else if (provider === _META) {
+    return {
+      prompt: requestBody.prompt,
+      temperature: requestBody.temperature || '',
+      top_p: requestBody.top_p || '',
+      max_tokens: requestBody.max_gen_len || ''
+    }
+  } else if (provider === _STABILITY) {
+    // TODO: request/response formats are different for image-based models. Defer for now
+    return {}
+  }
+  return {}
+}
+
+function extractTextAndResponseReason (response, provider, modelName, shouldSetChoiceIds) {
+  const body = JSON.parse(Buffer.from(response.body).toString('utf8'))
+  const textAndResponseReasons = []
+
+  try {
+    if (provider === 'AI21') {
```

Comment:
```suggestion
    if (provider === _AI21) {
```
let's match against the constant here instead (and remove the underscore once the constant name is changed)

Reply:
Yes !